cmake/modules/RootMacros.cmake

```diff
@@ -1513,7 +1513,7 @@ endfunction()
 #
 function(ROOT_ADD_GTEST test_suite)
   CMAKE_PARSE_ARGUMENTS(ARG "WILLFAIL" "" "COPY_TO_BUILDDIR;LIBRARIES;LABELS" ${ARGN})
-  include_directories(${CMAKE_CURRENT_BINARY_DIR} ${GTEST_INCLUDE_DIR} ${GMOCK_INCLUDE_DIR})
+  include_directories(${GTEST_INCLUDE_DIR})
```

Comment:
I am curious, why we still need `include_directories(${GTEST_INCLUDE_DIR})`? Is it possible to move it to `target_include_directories` as well?

Reply:
I get a build error I am having some difficulty debugging if I move the `${GTEST_INCLUDE_DIR}` into `target_include_directories()` - 
```
/home/reik/root/roottest/root/roofitstats/vectorisedPDFs/VectorisedPDFTests.h:23:10: fatal error: gtest/gtest.h: No such file or directory
   23 | #include "gtest/gtest.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.

```

Reply:
In which context is `VectorisedPDFTests.h` being compiled? Can you paste the full diagnostic?

Reply:
Sorry that the logs I posted earlier were not complete. 
This should be better - 
```
[100%] Building CXX object roottest/root/roofitstats/vectorisedPDFs/CMakeFiles/VectorisedPDFTests.dir/VectorisedPDFTests.cxx.o
In file included from /home/reik/root/roottest/root/roofitstats/vectorisedPDFs/VectorisedPDFTests.cxx:17:
/home/reik/root/roottest/root/roofitstats/vectorisedPDFs/VectorisedPDFTests.h:23:10: fatal error: gtest/gtest.h: No such file or directory
   23 | #include "gtest/gtest.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [roottest/root/roofitstats/vectorisedPDFs/CMakeFiles/VectorisedPDFTests.dir/build.make:80: roottest/root/roofitstats/vectorisedPDFs/CMakeFiles/VectorisedPDFTests.dir/VectorisedPDFTests.cxx.o] Error 1
make[2]: Leaving directory '/home/reik/root/root-build'
make[1]: *** [CMakeFiles/Makefile2:56449: roottest/root/roofitstats/vectorisedPDFs/CMakeFiles/VectorisedPDFTests.dir/all] Error 2
```

Reply:
Looks like this triggers a problem on the roottest side. We always used `ROOTTEST_ADD_UNITTEST_DIR` and it looks like the roofitstats uses `ROOT_ADD_GTEST` which is somehow reachable.

Adding @hageboeck to the discussion.

Reply:
My current understanding is that we add a special `VectorisedPDFTests` library consisting of `VectorisedPDFTests.cxx` which includes `VectorisedPDFTests.h` that depends on gtest.

I suspect if we merge all *cxx files into a single call to `ROOT_ADD_GTEST` it will create a single executable containing all tests plus the code being shared among them. The drawback is that we won't be able to do `-jN` and benefit from parallel tests.

Reply:
I don't think there's a need to merge all the files. Obviously, the `gtest/` folder is not where it's supposed to be. Presumably, that's since the folder `include_dir` was removed.
Suggest to use `make VERBOSE=1` to see why/where it's failing. I suspect that these two lines don't work, any more:
`/roottest/root/roofitstats/vectorisedPDFs/CMakeLists.txt`:
```
  8 add_library(VectorisedPDFTests STATIC VectorisedPDFTests.cxx)
  9 target_link_libraries(VectorisedPDFTests PUBLIC gtest ROOT::Gpad ROOT::RooFitCore ROOT::RooFit)
```
Possible fixes:
- For the `gtest` link target add an `INTERFACE` property that adds the gtest includes automatically to everything that links against it.
- Add an explicit `target_include_directories(VectorisedPDFTests ${GTEST_INCLUDE_DIR})`

Reply:
@hageboeck I applied your suggested second possible fix as roottest PR #489 and that seems to fix the earlier compilation error.

Reply:
Ok, I had a look. It's true that it works now, but I believe that this should have happened automatically. See root-project/roottest#486

Reply:
@reikdas I created the PR #5366. It hopefully fixes the missing includes for all targets that use gtest and gmock. Could you rebase onto that branch, and try it out?

Reply:
@hageboeck Rebasing on top of your PR seems to work :) Should I drop commits 0a1e18185cf0cabbd100141adeb455434b963db4 and 9b7d23f574ac25c1518a32e5229dbb61ee24162a from this PR and close https://github.com/root-project/roottest/pull/489 ?

Reply:
Yes, that sounds like the right thing to do. Let's merge #5366 first, though!

Reply:
#5366 is merged. You should be good to go now.